engine/src/main/java/org/terasology/rendering/dag/nodes/RefractiveReflectiveBlocksNode.java

```diff
@@ -278,39 +278,40 @@ public void process() {
     }
 
     @Override
-    public void update() {
-        lastUpdatedGBuffer.attachDepthBufferTo(refractiveReflectiveFbo);
-    }
-
-    @Override
     public void propertyChange(PropertyChangeEvent event) {
-        // This method is only called when oldValue != newValue.
-        if (event.getPropertyName().equals(RenderingConfig.NORMAL_MAPPING)) {
-            normalMappingIsEnabled = renderingConfig.isNormalMapping();
-            if (normalMappingIsEnabled) {
-                addDesiredStateChange(setTerrainNormalsInputTexture);
-                if (parallaxMappingIsEnabled) {
-                    addDesiredStateChange(setTerrainHeightInputTexture);
-                }
-            } else {
-                removeDesiredStateChange(setTerrainNormalsInputTexture);
-                if (parallaxMappingIsEnabled) {
-                    removeDesiredStateChange(setTerrainHeightInputTexture);
-                }
-            }
-        } else if (event.getPropertyName().equals(RenderingConfig.PARALLAX_MAPPING)) {
-            parallaxMappingIsEnabled = renderingConfig.isParallaxMapping();
-            if (normalMappingIsEnabled) {
-                if (parallaxMappingIsEnabled) {
-                    addDesiredStateChange(setTerrainHeightInputTexture);
+        if (event.getPropertyName().equals(DisplayResolutionDependentFBOs.PRE_FBO_REGENERATION)) {
+            refractiveReflectiveFbo.detachDepthBuffer();
+        } else if (event.getPropertyName().equals(DisplayResolutionDependentFBOs.POST_FBO_REGENERATION)) {
+            lastUpdatedGBuffer.attachDepthBufferTo(refractiveReflectiveFbo);
+        } else { // Else: One of the settings was changed, handle the event and refresh the task list.
+            // This method is only called when oldValue != newValue.
```

Comment:
Except there are no oldValue/newValue variables in use anywhere in the method. In this case I'd remove this line, unless we can make it more meaningful in this context.